### DISCUSSION TITLE
ElementWiseKernel and local_size/global_size

### DISCUSSION THREAD

**Discussion by craig-warren** [pyopencl/pyopencl#712] [General]
Hopefully naive question but how do you set `local_size` and `global_size` explicitly when using `ElementWiseKernel`? By default is something happening under the hood to choose these based on hardware and array sizes passed to the kernel?

**Comment by inducer** [comment]
There isn't a way, and there isn't one planned.

If you need that much control, just write your own kernel. If we tried to make `ElementwiseKernel` general enough to work with arbitrary work partitions, we would just make a mess of the code generation.

If you need a tool that can generate code for array computations that puts you in charge of group sizes, https://github.com/inducer/loopy may be worth a look.

**Reply by craig-warren** [reply]
@inducer understood, thanks.
